frontend/src/store/wallet.ts

```diff
@@ -613,11 +613,16 @@ const hasSetPublicKeysLegacy = async (name: string, provider: Provider) => {
 
 // Helper method to check if user has registered public keys in the StealthKeyRegistry
 async function getRegisteredStealthKeys(account: string, provider: Provider) {
-  try {
-    const stealthPubKeys = await utils.lookupRecipient(account, provider); // throws if no keys found
-    return stealthPubKeys;
-  } catch (err) {
-    window.logger.warn(err);
-    return null;
+  let retryCounter = 0;
+  while (retryCounter < 3) {
```

Comment:
1. Should we add timeout between tries?
2. I think we've discussed this, maybe we should try to query public providers if the first one fails. But maybe that won't work either because public providers are prob heavily rate limited.
3. I think the long term solution to this is querying a subgraph instead, which we should have an issue for, if not we should create one.

Reply:
Not sure option 3 is something we'd want to expend effort on in Umba v1, especially as we've seen this failure pretty infrequently, correct?

Reply:
Added a delay of 2 seconds for each of the 3 retries: https://github.com/ScopeLift/umbra-protocol/pull/655/commits/2f0d54b76f472a0230787bf69eea43a3a09450de